plugin/v2/plugin.go

```diff
@@ -120,15 +125,37 @@ func (p *Plugin) InitPlugin(libRoot string) error {
 
 // Set is used to pass arguments to the plugin.
 func (p *Plugin) Set(args []string) error {
-	m := make(map[string]string, len(args))
-	for _, arg := range args {
-		i := strings.Index(arg, "=")
-		if i < 0 {
-			return fmt.Errorf("No equal sign '=' found in %s", arg)
+	p.Lock()
+	defer p.Unlock()
+
+	if p.PluginObj.Enabled {
+		return fmt.Errorf("cannot set on an active plugin, disable plugin before setting")
+	}
+
+	for _, s := range newSettables(args) {
+		// range over all the envs in the manifest
+		for _, env := range p.PluginObj.Manifest.Env {
+			// found the env in the manifest
+			if env.Name == s.name {
+				// is it settable ?
+				if !s.isSettable(allowedSettableFieldsEnv, env.Settable) {
```

Comment:
`isSettable` is happening in a O(n*n) and in itself has another for loop running. that makes the entire loop O(n^3). Lets think about revisiting this.